<launch>
    <!-- Drone id -->
    <arg name="drone_id"/>
    <arg name="drone_id_ns" value="drone$(arg drone_id)"/>
    <arg name="num_drones"/>
    <arg name="init_x"/>
    <arg name="init_y"/>
    <arg name="init_z"/>
    <!-- size of map, change the size inflate x, y, z according to your application -->
    <arg name="map_size_x"/>
    <arg name="map_size_y"/>
    <arg name="map_size_z"/>
    <!-- Input/output topics -->
    <arg name="command_topic" default="mavros/setpoint_raw/local" />
    <arg name="odometry_topic" default="mavros/local_position/odom" />
    <arg name="pose_topic" default="mavros/local_position/pose" />
    <arg name="depth_img_topic" default="camera/depth/image_raw" />
    <arg name="cloud_topic" default="camera/depth/points" />
    <arg name="POSE_TYPE" />
    <arg name="SENSOR_TYPE" />
    <!-- number of moving objects -->
    <arg name="obj_num" value="10" />
    <!-- drone_origin_frame: For PX4 SITL, leave it as default. 
    For other forms of simulation, might have to change it -->
    <arg name="drone_origin_frame" default="$(arg drone_id_ns)_origin" />

    <!-- Ego Planner params -->
    <include file="$(find gestelt_bringup)/launch/advanced_param.xml">
        <arg name="drone_origin_frame" value="$(arg drone_origin_frame)"/>

        <arg name="drone_id" value="$(arg drone_id)"/>
        <arg name="num_drones" value="$(arg num_drones)" />

        <arg name="map_size_x" value="$(arg map_size_x)"/>
        <arg name="map_size_y" value="$(arg map_size_y)"/>
        <arg name="map_size_z" value="$(arg map_size_z)"/>
        <arg name="obj_num_set" value="$(arg obj_num)" />

        <arg name="POSE_TYPE" value="$(arg POSE_TYPE)" />
        <arg name="SENSOR_TYPE" value="$(arg SENSOR_TYPE)" />
        
        <!-- camera pose: transform of camera frame in the world frame -->
        <!-- depth topic: depth image, 640x480 by default -->
        <!-- cloud_topic: Point cloud topic   -->
        <arg name="depth_topic" value="/$(arg drone_id_ns)/$(arg depth_img_topic)"/>
        <arg name="cloud_topic" value="/$(arg drone_id_ns)/$(arg cloud_topic)"/>

        <arg name="pose_topic" value="/$(arg drone_id_ns)/$(arg pose_topic)"/>
        <arg name="odometry_topic" value="/$(arg drone_id_ns)/$(arg odometry_topic)"/>

        <!-- maximum velocity, acceleration and jerk the drone will reach -->
        <arg name="max_vel" value="1.5" />
        <arg name="max_acc" value="6.0" />
        <arg name="max_jer" value="20.0" />
        
        <!-- thresh_no_replan_meter: Replan only if current position and local target exceeds this parameter -->
        <arg name="thresh_no_replan_meter" value="0.25" />
        <!-- thresh_replan_time: Replanning period -->
        <arg name="thresh_replan_time" value="0.5" />
        <!--always set to 1.5 times grater than sensing horizon-->
        <!-- sensing horizon here refers to the sensor range. e.g. about 4.5-5m for realsense d435 -->
        <arg name="planning_horizon" value="7.5" />
        <!-- TODO: What is this param? -->
        <arg name="use_distinctive_trajs" value="false" />
        <!-- 1: use 2D Nav Goal to select goal  -->
        <!-- 2: Assign waypoints via a topic  -->
        <arg name="waypoint_type" value="2" />
    </include>

    <include file="$(find gestelt_bringup)/launch/trajectory_server.xml">
        <arg name="drone_id" value="$(arg drone_id)"/>
        <arg name="server_mode" value="0" />
        <!-- drone_origin_frame: For PX4 SITL, leave it as default. 
        For other forms of simulation, might have to change it -->
        <arg name="drone_origin_frame" value="$(arg drone_origin_frame)" />
        <!-- size of map, change the size inflate x, y, z according to your application -->
        <arg name="map_size_x" value="$(arg map_size_x)"/>
        <arg name="map_size_y" value="$(arg map_size_y)"/>
        <arg name="map_size_z" value="$(arg map_size_z)"/>
        <!-- Input/output topics -->
        <arg name="command_topic"   value="$(arg command_topic)" />
        <arg name="odometry_topic"   value="$(arg odometry_topic)" />
        <arg name="pose_topic"   value="$(arg pose_topic)" />
    </include>

</launch>
